lua/wire/zvm/zvm_opcodes.lua

```diff
@@ -1221,6 +1221,16 @@ ZVM.OpcodeTable[141] = function(self) --EXTRETPA
 end
 
 
+ZVM.OpcodeTable[150] = function(self) -- STERR
+    self:Dyn_Emit("$L P = $2")
+    self:Dyn_Emit("P = P * (10^math.floor(-math.log10(math.abs(P)+1e-12)-1))")
+    self:Dyn_Emit("VM:SignalError(math.floor($1)+P)")
```

Comment:
Could you add comments for what $1 and $2 represent and what this equation is doing? I also feel like this can be done more optimally without exponentials and logarithms.

Reply:
In the microcode strings that all CPU instructions use, there are a few shorthands that are added in a preprocessing step between Dyn_Emit and Emit

$1 is a variable representing operand 1 of the instruction
$2 is a variable representing operand 2 of the instruction

So in the case of

MOV R1,65

$1 is R1
$2 is 65

The code for generating P(standing for "parameter" as in the parameter for the error), was adapted from the [original code](https://github.com/wiremod/wire-cpu/blob/494d35fd9a3471776454d839b0f4709b49949c4c/lua/wire/zvm/zvm_features.lua#L663), to try and keep the output similar to how the error output is generated when an interrupt occurs

If you feel the error parameter generation func should change, I'm open to testing any suggestions you may have

Reply:
Ah yeah I remember now, so it's putting the error's parameter in the decimal of the number. A code of 5, 12 would be 5.12

Reply:
Could you move the equation to a `MakeErrorCode(code, param)` and make both of those code locations use it?

Reply:
Done, do you still want the comments for $1 and $2?

Reply:
No that's fine.

For the function I was wanting to to take two parameters. That way your code could just be:
`self:Dyn_Emit("VM:SignalError(VM:MakeErrorCode($1, $2))")`

Reply:
Oh I misunderstood, corrected.